File: PyzeWatchOSInAppObjC/PyzeWatchOSInAppObjC WatchKit Extension/InterfaceController.m

#import <Pyze/Pyze.h>

#import "InterfaceController.h"
#import "InAppMessageManager.h"

@interface InterfaceController()

@property (unsafe_unretained, nonatomic) IBOutlet WKInterfaceButton *unreadbutton;

@end


@implementation InterfaceController

- (void)awakeWithContext:(id)context {
    [super awakeWithContext:context];

    [Pyze countNewUnFetched:^(NSInteger count) {
        dispatch_async(dispatch_get_main_queue(), ^{
            [self.unreadbutton setTitle:[NSString stringWithFormat:@"Show Unread (%d)",count]];
        });
    }];
}

- (void)willActivate {
    // This method is called when watch view controller is about to be visible to user
    [super willActivate];
}

- (void)didDeactivate {
    // This method is called when watch view controller is no longer visible
    [super didDeactivate];
}

#pragma mark - Action methods

- (IBAction)showUnread {
    [[InAppMessageManager sharedManager] fetchHeadersForType:PyzeInAppTypeUnread completionHandler:^{
        [self pushControllerWithName:@"InAppMessageUIController" context:nil];
    }];
}

- (IBAction)showRead {
    [[InAppMessageManager sharedManager] fetchHeadersForType:PyzeInAppTypeRead completionHandler:^{
        [self pushControllerWithName:@"InAppMessageUIController" context:nil];
    }];
}

- (IBAction)showAll {
    [[InAppMessageManager sharedManager] fetchHeadersForType:PyzeInAppTypeAll completionHandler:^{
        [self pushControllerWithName:@"InAppMessageUIController" context:nil];
    }];
}


@end

File: PyzeWatchOSInApp/PyzeWatchOSInApp WatchKit Extension/InterfaceController.swift
import WatchKit
import Foundation
import Pyze

class InterfaceController: WKInterfaceController {
    
    
    @IBOutlet var unreadButton: WKInterfaceButton!
    
    // MARK: - View life cycle
    
    override func awake(withContext context: Any?) {
        super.awake(withContext: context)
        
        InAppMessageManager.sharedManager.countNewUnFetched { (count) in
            DispatchQueue.main.async {
                self.unreadButton.setTitle("Unread (\(count))")
            }
        }
    }
    
    override func willActivate() {
        super.willActivate()
    }
    
    override func didAppear() {
        super.didAppear()
    }
    
    
    override func didDeactivate() {
        super.didDeactivate()
    }
    
    // MARK: - Action methods
    
    @IBAction func showUnread() {
        InAppMessageManager.sharedManager.fetchHeaders(type: .typeUnread, completionhandler: {
            
            DispatchQueue.main.async {
                self.pushController(withName: "InAppMessageUIController", context:nil)
            }
            
        })
    }
    
    @IBAction func showRead() {
        InAppMessageManager.sharedManager.fetchHeaders(type: .typeRead, completionhandler: {
            
            DispatchQueue.main.async {
                self.pushController(withName: "InAppMessageUIController", context:nil)
            }
            
        })
    }
    
    @IBAction func showAll() {
        InAppMessageManager.sharedManager.fetchHeaders(type: .typeAll, completionhandler: {
            
            DispatchQueue.main.async {
                self.pushController(withName: "InAppMessageUIController", context:nil)
            }
        })
    }
}
